lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1538 @@
+        $defaultConfigFiles = @('rootwrap.d', 'api-paste.ini', 'cells.json',
+                                'rootwrap.conf')
+        foreach ($config in $defaultConfigFiles) {
+            Copy-Item -Recurse -Force "$openstackBuild\nova\etc\nova\$config" $CONFIG_DIR
+        }
+        Copy-Item -Force (Join-Path (Get-TemplatesDir) "policy.json") $CONFIG_DIR
+        Copy-Item -Force (Join-Path (Get-TemplatesDir) "interfaces.template") $CONFIG_DIR
+    }
+    # Networking-Hyperv
+    if ($buildFor -eq "openstack/networking-hyperv") {
+        $neutronBin = (Get-CharmServices)['neutron']['binary']
+        if (!(Test-Path $neutronBin)) {
+            Throw "$neutronBin was not found."
+        }
+    }
+}
+
+function Get-ComputeDriver {
+    if ($buildFor -eq "openstack/compute-hyperv") {
+        $compute_driver = "hyperv.nova.driver.HyperVDriver"
+    }
+    else {
+        $compute_driver = "hyperv.driver.HyperVDriver"
+    }
+    return $compute_driver
+}
+
+function Install-NetworkType {
+    $buildFor = Get-JujuCharmConfig -Scope 'zuul-project'
+    $networkType = Get-JujuCharmConfig -Scope 'network-type'
+    if (($networkType -eq 'ovs') -and ($buildFor -eq 'openstack/networking-hyperv')) {
+        Throw "'$networkType' cannot be used when building for '$buildFor'"
+    }
+    if ($networkType -eq 'hyperv') {
+#        Install-NetworkingHyperV
+        Install-Projects "openstack/networking-hyperv"
+    }
+    elseif ($networkType -eq 'ovs') {
+        Check-OVSPrerequisites
+        Enable-OVS
+        Ensure-InternalOVSInterfaces
+    }
+    else {
+        Throw "Wrong network type config: '$networkType'"
+    }
+}
+
+#function Install-NetworkingHyperV {
+#    Write-JujuLog "Installing networking-hyperv"
+#
+#    $openstackBuild = Join-Path $BUILD_DIR "openstack"
+#    Start-ExecuteWithRetry {
+#        Install-OpenStackProjectFromRepo "$openstackBuild\networking-hyperv"
+#    }
+#    $neutronBin = (Get-CharmServices)['neutron']['binary']
+#    if (!(Test-Path $neutronBin)) {
+#        Throw "$neutronBin was not found."
+#    }
+#}
+
+
+function Install-OVS {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$InstallerPath
+    )
+
+    Write-JujuInfo "Running OVS install"
+    $ovs = Get-ManagementObject -Class Win32_Product | Where-Object {$_.Name -match "open vswitch"}
+    if ($ovs){
+        Write-JujuInfo "OVS is already installed"
+        return $true
+    }
+
+    $hasInstaller = Test-Path $InstallerPath
+    if($hasInstaller -eq $false){
+        $InstallerPath = Get-OVSInstaller
+    }
+
+    Write-JujuInfo "Installing from $InstallerPath"
+    $ret = Start-Process -FilePath msiexec.exe -ArgumentList "INSTALLDIR=`"$OVS_DIR`"","/qb","/l*v","$env:APPDATA\ovs-log.txt","/i","$InstallerPath" -Wait -PassThru
+    if($ret.ExitCode) {
+        Throw "Failed to install OVS: $LASTEXITCODE"
+    }
+    Remove-Item $InstallerPath
+    return $true
+}
+
+
+function Import-OVSCertificate {
```

Comment:
This is needed for OVS-CI . The generated OVS msi has a self-signed certificate that get's imported here.